SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc

```diff
@@ -345,69 +321,65 @@ void Phase2TrackerDigitizerAlgorithm::drift(const PSimHit& hit,
-    SegY = ionization_points[i].y();
-    SegZ = ionization_points[i].z();
+  for (auto const& val : ionization_points) {
+    // position
+    float SegX = val.x(), SegY = val.y(), SegZ = val.z();
 
     // Distance from the collection plane
     // DriftDistance = (moduleThickness/2. + SegZ); // Drift to -z
     // Include explixitely the E drift direction (for CMS dir_z=-1)
-    DriftDistance = moduleThickness / 2. - (dir_z * SegZ);  // Drift to -z
 
-    if (DriftDistance < 0.)
-      DriftDistance = 0.;
-    else if (DriftDistance > moduleThickness)
-      DriftDistance = moduleThickness;
+    // Distance between charge generation and collection
+    float driftDistance = moduleThickness / 2. - (dir_z * SegZ);  // Drift to -z
+
+    if (driftDistance < 0.)
+      driftDistance = 0.;
+    else if (driftDistance > moduleThickness)
+      driftDistance = moduleThickness;
 
     // Assume full depletion now, partial depletion will come later.
-    float XDriftDueToMagField = DriftDistance * TanLorenzAngleX;
-    float YDriftDueToMagField = DriftDistance * TanLorenzAngleY;
+    float XDriftDueToMagField = driftDistance * TanLorenzAngleX;
+    float YDriftDueToMagField = driftDistance * TanLorenzAngleY;
 
     // Shift cloud center
     float CloudCenterX = SegX + XDriftDueToMagField;
     float CloudCenterY = SegY + YDriftDueToMagField;
 
     // Calculate how long is the charge drift path
-    DriftLength = std::sqrt(DriftDistance * DriftDistance + XDriftDueToMagField * XDriftDueToMagField +
-                            YDriftDueToMagField * YDriftDueToMagField);
+    // Actual Drift Lentgh
+    float driftLength =
+        std::sqrt(std::pow(driftDistance, 2) + std::pow(XDriftDueToMagField, 2) + std::pow(YDriftDueToMagField, 2));
 
     // What is the charge diffusion after this path
     // Sigma0=0.00037 is for 300um thickness (make sure moduleThickness is in [cm])
-    Sigma = std::sqrt(DriftLength / moduleThickness) * (Sigma0 * moduleThickness / 0.0300);
+    float Sigma = std::sqrt(driftLength / moduleThickness) * Sigma0_ * moduleThickness / 0.0300;
     // D.B.: sigmaCoeff=0 means no modulation
-    if (SigmaCoeff)
-      Sigma *= (SigmaCoeff * cos(SegX * M_PI / stripPitch) * cos(SegX * M_PI / stripPitch) + 1);
+    if (SigmaCoeff_)
+      Sigma *= (SigmaCoeff_ * std::pow(cos(SegX * M_PI / stripPitch), 2) + 1);
     // NB: divided by 4 to get a periodicity of stripPitch
 
     // Project the diffusion sigma on the collection plane
-    Sigma_x = Sigma / CosLorenzAngleX;
-    Sigma_y = Sigma / CosLorenzAngleY;
+    float Sigma_x = Sigma / CosLorenzAngleX;
+    float Sigma_y = Sigma / CosLorenzAngleY;
 
     // Insert a charge loss due to Rad Damage here
-    float energyOnCollector = ionization_points[i].energy();  // The energy that reaches the collector
+    float energyOnCollector = val.energy();  // The energy that reaches the collector
 
     // pseudoRadDamage
-    if (pseudoRadDamage >= 0.001) {
+    if (pseudoRadDamage_) {
```

Comment:
Is this change intentional? The behavior is different from the previous version of this line.